File: python/lld/state-machines/atm/idle_state.py
from atm_state import ATMState
from has_card_state import HasCardState


class IdleState(ATMState):
    """Only legal move is to insert a card."""

    def name(self) -> str:
        return "IDLE"

    def insert_card(self):
        print("  -> card accepted")
        self.atm.set_state(HasCardState(self.atm))

File: java/lld/state-machines/atm/IdleState.java
// Nothing in the machine; the only legal move is to insert a card.
public class IdleState extends ATMState {
    public IdleState(ATM atm) { super(atm); }
    public String name() { return "IDLE"; }

    @Override public void insertCard() {
        System.out.println("  -> card accepted");
        atm.setState(new HasCardState(atm));
    }
}